core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -1486,9 +1486,25 @@ public void delete(AlluxioURI path, DeleteOptions options) throws IOException,
         LockedInodePath inodePath =
             mInodeTree.lockInodePath(lockingScheme.getPath(), lockingScheme.getMode());
         FileSystemMasterAuditContext auditContext =
-            createAuditContext("delete", path, null, inodePath.getInodeOrNull())) {
+            createAuditContext("delete", path, null, inodePath.getInodeOrNull());
+        LockedInodePathList children = !options.isRecursive() ? null
```

Comment:
do we expect a performance overhead for reentrant locking?

Reply:
if `children` is `null`, does the try-with-resources still work correctly with the `null`?

Reply:
@yuzhu should have context on performance implications.

Reply:
yeah, try-with-resources works with null (tested it)